bridges/sloghandler/handler.go

```diff
@@ -0,0 +1,360 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package sloghandler provides a bridge between the [log/slog] and
+// OpenTelemetry logging.
+package sloghandler // import "go.opentelemetry.io/contrib/bridges/sloghandler"
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"slices"
+	"sync"
+
+	"go.opentelemetry.io/otel/log"
+	"go.opentelemetry.io/otel/log/noop"
+)
+
+const (
+	bridgeName = "go.opentelemetry.io/contrib/bridge/sloghandler"
+	// TODO: hook this into the release pipeline.
+	bridgeVersion = "0.0.1-alpha"
+)
+
+type config struct{}
+
+// Option configures a [Handler].
+type Option interface {
+	apply(config) config
+}
+
+// Handler is an [slog.Handler] that sends all logging records it receives to
+// OpenTelemetry.
+type Handler struct {
+	// Ensure forward compatibility by explicitly making this not comparable.
+	noCmp [0]func() //nolint: unused  // This is indeed used.
+
+	attrs  *kvBuffer
+	group  *group
+	logger log.Logger
+}
+
+// Compile-time check *Handler implements slog.Handler.
+var _ slog.Handler = (*Handler)(nil)
+
+// New returns a new [Handler] to be used as an [slog.Handler].
+func New(lp log.LoggerProvider, _ ...Option) *Handler {
+	if lp == nil {
+		// Do not panic.
+		lp = noop.NewLoggerProvider()
+	}
```

Comment:
I was leaning more to requiring the `LoggerProvider` passed as an argument directly given it is required. If the user wants to use the global or a custom created one they will be confronted with the requirement and decide prior to creating the bridge. It will avoid [this](https://stackoverflow.com/questions/78020154/go-labstack-echo-and-opentelemetry-doesnt-work) type of confusion.

Reply:
I like your proposal. This behavior could be documented. In future, we should NOT change to use the global logger provider when `nil` is passed.

Feel free to resolve the comment.